### PR TITLE
Add poster attribute to VideoObjectModel

### PR DESCRIPTION
**Pull request by ffarfan**
We add a "poster" attribute to the VideoObjectModel to marshall poster images
to display when the video is about to play.

https://phabricator.endlessm.com/T14156

### CODE REVIEW

**Comment by sgnn7** [comment]
LGTM but might be good for @magcius and/or @wgreenberg to take a look too

File: js/search/mediaObjectModel.js [hunk 1]
```diff
@@ -43,7 +43,17 @@ const MediaObjectModel = new Lang.Class({
         'width': GObject.ParamSpec.uint('width',
             'Width', 'The width of the media in pixels',
             GObject.ParamFlags.READWRITE | GObject.ParamFlags.CONSTRUCT_ONLY,
-            0, GLib.MAXUINT32, 0)
+            0, GLib.MAXUINT32, 0),
+
+        /**
+         * Property:poster-uri
+         * URI of the the video's poster image
```
Comment (rmacqueen) [inline js/search/mediaObjectModel.js]: Does this need to be on `MediaObjectModel` as opposed to just on `VideoObjectModel`?

**Comment by ffarfan** [comment]
@rmacqueen revised now!

File: js/search/mediaObjectModel.js [hunk 2]
```diff
@@ -63,6 +63,9 @@ const MediaObjectModel = new Lang.Class({
 
         if (json_ld.hasOwnProperty('width'))
             props.width = parseInt(json_ld.width);
+
+        if (json_ld.hasOwnProperty('poster'))
```
Comment (rmacqueen) [inline js/search/mediaObjectModel.js]: Shouldn't this now move down into `VideoObjectModel` too?

**Comment by rmacqueen** [comment]
CR PASS

**Pull request merged**